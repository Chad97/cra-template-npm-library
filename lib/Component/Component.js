import React from 'react';
import classes from './Component.module.css';

const Component = () => (
  <div className={classes.component}>Component hello world</div>
);

const a = {};

export default Component;
