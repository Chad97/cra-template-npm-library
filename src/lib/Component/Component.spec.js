import React from 'react';
import {render} from '@testing-library/react';
import Component from './Component';

describe('lib > Component', () => {
  it('renders without crashing', () => {
    /**
     * `asFragment`:
     * @see https://testing-library.com/docs/react-testing-library/api#asfragment
     */
    const {asFragment} = render(<Component />);

    /**
     * Basic snapshot test to make sure, that rendered component
     * matches expected footprint.
     */
    expect(asFragment()).toMatchSnapshot();
  });
});
