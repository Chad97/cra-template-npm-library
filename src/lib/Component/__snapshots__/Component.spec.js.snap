// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib > Component renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="component"
  >
    Component hello world
  </div>
</DocumentFragment>
`;
